Jenkinsfile

```diff
@@ -0,0 +1,256 @@
+#!/usr/bin/env groovy
+@Library('apm@v1.0.6') _
+
+import co.elastic.matrix.*
+import groovy.transform.Field
+
+/**
+This is the parallel tasks generator,
+it is need as field to store the results of the tests.
+*/
+@Field def pythonTasksGen
+
+pipeline {
+  agent any
+  environment {
+    BASE_DIR="src/github.com/elastic/apm-agent-python"
+    PIPELINE_LOG_LEVEL='INFO'
+    NOTIFY_TO = credentials('notify-to')
+    JOB_GCS_BUCKET = credentials('gcs-bucket')
+  }
+  options {
+    timeout(time: 1, unit: 'HOURS')
+    buildDiscarder(logRotator(numToKeepStr: '20', artifactNumToKeepStr: '20', daysToKeepStr: '30'))
+    timestamps()
+    ansiColor('xterm')
+    disableResume()
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+  }
+  triggers {
+    issueCommentTrigger('.*(?:jenkins\\W+)?run\\W+(?:the\\W+)?tests(?:\\W+please)?.*')
+  }
+  parameters {
+    booleanParam(name: 'Run_As_Master_Branch', defaultValue: false, description: 'Allow to run any steps on a PR, some steps normally only run on master branch.')
+    booleanParam(name: 'doc_ci', defaultValue: true, description: 'Enable build docs.')
+  }
+  stages {
+    stage('Initializing'){
+      agent { label 'docker && linux && immutable' }
+      options { skipDefaultCheckout() }
+      environment {
+        HOME = "${env.WORKSPACE}"
+        PATH = "${env.PATH}:${env.WORKSPACE}/bin"
+        ELASTIC_DOCS = "${env.WORKSPACE}/elastic/docs"
+      }
+      stages {
+        /**
+        Checkout the code and stash it, to use it on other stages.
+        */
+        stage('Checkout') {
+          steps {
+            deleteDir()
+            gitCheckout(basedir: "${BASE_DIR}")
+            stash allowEmpty: true, name: 'source', useDefaultExcludes: false
+            dir("${BASE_DIR}"){
+              sh "git log origin/${env.CHANGE_TARGET}...${env.GIT_SHA}"
+            }
+          }
+        }
+        /**
+        Build the project from code..
+        */
+        stage('Build') {
```

Comment:
I don't think "Build" is the right name for this stage. `isort` and `black` both check code style. "Lint" maybe?

Reply:
ok